src/mapping.ts

```diff
@@ -1,410 +1,477 @@
-    .plus(exerciseValueUSD)
-    .minus(underlyingValueUSD);
-
-  contract.save();
-
-  let underlyingAsset = loadOrInitializeToken(
-    Address.fromString(underlyingAssetAddress as string)
+  // handle daily metrics
+  const tokenDaily = fetchDailyTokenMetrics(
+    event.params.asset.toHexString(),
+    event.block.timestamp
   );
-  underlyingAsset.totalValueLocked = underlyingAsset.totalValueLocked.minus(
-    underlyingAmount
+  tokenDaily.notionalVolFeesSwept = tokenDaily.notionalVolFeesSwept.plus(
+    event.params.amount
   );
-  underlyingAsset.totalValueLockedUSD = underlyingAsset.totalValueLockedUSD.minus(
-    underlyingValueUSD
+  tokenDaily.notionalVolFeesSweptUSD = tokenDaily.notionalVolFeesSweptUSD.plus(
+    feeValueUSD
   );
-  underlyingAsset.save();
+  tokenDaily.save();
 
-  let exerciseAsset = loadOrInitializeToken(
-    Address.fromString(exerciseAssetAddress as string)
-  );
-  exerciseAsset.totalValueLocked = exerciseAsset.totalValueLocked.plus(
-    exerciseAmount
-  );
-  exerciseAsset.totalValueLockedUSD = exerciseAsset.totalValueLockedUSD.plus(
-    exerciseValueUSD
+  const dailyOSEMetrics = fetchDailyOSEMetrics(event.block.timestamp);
+  dailyOSEMetrics.notionalVolFeesSweptUSD = dailyOSEMetrics.notionalVolFeesSweptUSD.plus(
+    feeValueUSD
   );
-  exerciseAsset.save();
-
-  let dayData = updateValoremDayData(event);
-
-  dayData.volumeUSD = dayData.volumeUSD.plus(underlyingValueUSD);
+  dailyOSEMetrics.save();
+}
 
-  dayData.save();
+// https://github.com/OpenZeppelin/openzeppelin-subgraphs
+export function handleTransferSingle(event: TransferSingleEvent): void {
+  let contract = fetchERC1155(event.address.toHexString());
+  let operator = fetchAccount(event.params.operator.toHexString());
+  let from = fetchAccount(event.params.from.toHexString());
+  let to = fetchAccount(event.params.to.toHexString());
 
-  let underlyingDayData = updateTokenDayData(underlyingAsset, event);
-  underlyingDayData.volume = underlyingDayData.volume.plus(underlyingAmount);
-  underlyingDayData.volumeUSD = underlyingDayData.volumeUSD.plus(
-    underlyingValueUSD
+  registerTransfer(
+    event,
+    "",
+    contract,
+    operator,
+    from,
+    to,
+    event.params.id,
+    event.params.amount
   );
-  underlyingDayData.save();
 
-  updateTokenDayData(exerciseAsset, event);
+  /**
+   * Valorem Extension
+   * Checks if to/from addresses are not one of Valorem's or the Zero Address
+   * If neither are, records the amounts transferred in daily metrics
+   */
+  // get params
+  const fromAddress = event.params.from.toHexString();
+  const toAddress = event.params.to.toHexString();
+  const tokenId = event.params.id;
+  const amount = event.params.amount;
+
+  // check transfer isn't an OSE Write/Exercise/Redeem event (mint/OSE is receiver/burn)
+  const transferIsExternal =
+    fromAddress != ZERO_ADDRESS &&
+    toAddress != ZERO_ADDRESS &&
+    toAddress != event.address.toHexString();
```

Comment:
Hmm, have to think this through, but what is the scenario where an 1155 is getting transferred _to_ the OSE contract itself?

Reply:
Honestly, there shouldn't be. And it's not like the OSE can transfer them out either. Might remove this check.

Reply:
removed OSE check here: https://github.com/valorem-labs-inc/valorem-subgraph/commit/59eefb0dc693d6b4f3a5d88582dcd7335a26a7a4